### ISSUE TITLE
Broken regex causing incorrect highlighting

### ISSUE THREAD

**Issue by davidhcefx** [serialhex/nano-highlight#51]
When trying out this project, I found several issues:

1. Since I added `include /path/to/nano-highlight/*.nanorc` to my nanorc, the highlighting of `.nanorc` had been changed. However, even I removed `nanorc.nanorc`, it is still happening. To my surprise, `rcfiles-new.nanorc` was the culprit:
https://github.com/serialhex/nano-highlight/blob/672aad96cd2c02f2c4e601d3dc1feed4eda8ffae/rcfiles-new.nanorc#L9-L11

Notice the broken regex `rc$`, which match not only `.bashrc` but also `nanorc`. Furthermore, `.gtk` would be matched instead of `.gtkrc`. The same issue might also exist in other rc files.

2. `bash.nanorc` is colliding its definition of `\.sh$` with `sh.nanorc`. It does not work as a *fallback method*, but instead a *complete overwrite*. More worst, the order of which one overwrite which is ambiguous, so it's definitely not a good idea.

**Comment by serialhex** [comment]
Hmm, I had not noticed this.  I'm not sure exactly what I want to do about this, but it needs to be fixed.  If you have any suggestions let me know.  It is probably going to be a while before I have a chance to take a look & fix this myself.

**Comment by j-f1** [comment]
A couple of possibilities:

```nanorc
 syntax "rcfiles" "^(\.?(ba|c|z)sh|input|xinit|eix|\.?wminet|gtk)rc$" 
 syntax "rcfiles" "^\.?[^.]+rc$" 
```

**Comment by davidhcefx** [comment]
Thanks for your prompt reply! For the possible fixes, I would prefer the first one, but shouldn't we constraint to `.inputrc$` instead of `inputrc$`? eg.

```nanorc
syntax "rcfiles" "^\.?((ba|c|z)sh|input|xinit|eix|wminet|gtk)rc$" 
```

**Comment by davidhcefx** [comment]
By the way, don't place a starting anchor to the regex, **because it is a full path match**!

**Comment by serialhex** [comment]
Fixed by PR#52
